runtime/src/impls.rs

```diff
@@ -35,6 +36,37 @@ use sp_runtime::{
 	DispatchError, Fixed64,
 };
 
+type NegativeImbalance = <<Runtime as crml_transaction_payment::Trait>::Currency as Currency<
+	<Runtime as frame_system::Trait>::AccountId>>::NegativeImbalance;
+type PositiveImbalance = <<Runtime as crml_transaction_payment::Trait>::Currency as Currency<
+	<Runtime as frame_system::Trait>::AccountId>>::PositiveImbalance;
+
+type CennzxSpot<T> = crml_cennzx_spot::Module<T>;
+type Contracts<T> = pallet_contracts::Module<T>;
+type GenericAsset<T> = pallet_generic_asset::Module<T>;
+type Staking<T> = pallet_staking::Module<T>;
+
+pub struct Validators;
+impl OnUnbalanced<NegativeImbalance> for Validators
+{
+	fn on_nonzero_unbalanced(amount: NegativeImbalance) {
+		let validators = Staking::<Runtime>::current_elected();
+		if validators.len() > 0 {
+			let per_validator_reward:Balance = amount.peek() / (validators.len() as Balance);
+
+			let mut total_imbalance = PositiveImbalance::zero();
+			for v in &validators {
+				let payout = <GenericAsset<Runtime> as MultiCurrencyAccounting>::deposit_creating(
+					&v,
+					Some(amount.asset_id().into()),
+					per_validator_reward);
+				total_imbalance.subsume(payout);
+			}
+
+		}
+	}
```

Comment:
Suggest keeping track of any unrewarded imbalance and burning it at the end of this function. 

This will also take care case where validators.len() == 0.

Reply:
Spoke with Roy, this is handled